datafusion/sql/src/parser.rs

```diff
@@ -438,12 +440,26 @@ impl<'a> DFParser<'a> {
         &self,
         expected: &str,
         found: TokenWithSpan,
-    ) -> Result<T, ParserError> {
-        parser_err!(format!("Expected {expected}, found: {found}"))
+    ) -> Result<T, DataFusionError> {
+        let sql_parser_span = found.span;
+        parser_err!(format!(
+            "Expected: {expected}, found: {found}{}",
+            found.span.start
+        ))
+        .map_err(|e| {
+            let e: DataFusionError = e.into();
```

Comment:
Here is an alternate syntax that I find more explicit, but totally uneeded

```suggestion
            let e = DataFusionError::from(e);
```